src/geohash_helper.c

```diff
@@ -233,3 +240,15 @@ int geohashGetDistanceIfInRadiusWGS84(double x1, double y1, double x2,
                                       double *distance) {
     return geohashGetDistanceIfInRadius(x1, y1, x2, y2, radius, distance);
 }
+
+/* Judge whether a point is in the axis-aligned rectangle.
+ * bounds : see geohash.h GeoShape::bounds
+ * x1, y1 : the center of the box
+ * x2, y2 : the point to be searched
+ */
+int geohashGetDistanceIfInRectangle(double *bounds, double x1, double y1,
+                                    double x2, double y2, double *distance) {
+    if (x2 < bounds[0] || x2 > bounds[2] || y2 < bounds[1] || y2 > bounds[3]) return 0;
```

Comment:
something still doesn't feel right here.

the `geohashGetDistanceIfInRadius` makes sense, you get two points "1" and "2" to measure the distance between them, and return it only if smaller than a "radius" distance.

here we get two points ("1" and "2"), and i would imagine that also a width/2 and height/2, to return the the distance between the two points only if smaller than width/2 horizontally, and height/2 vertically.
but instead we get "bounds" which is a 4 element long / lat thing.

also, looking at `geohashBoundingBox` i see that `bounds` already has shape->xy embedded in it, so these are not just w/2 and h/2.

lastly, i remind you that our `bounds` is bloated (using a fake radius that's generated by max(w/2,h/2), so checking here against `bounds` is probably a bug.

my advise: this function should probably get width and height as input instead of bounds.

Reply:
all coordinates (x, y, bounds) are in long/lat
all distances (width, height, radius) are in meters.
If we use width and height as parameters, we need to recalculate the latitude and longitude (meters to degrees), to avoid calculation, we pass bounds.